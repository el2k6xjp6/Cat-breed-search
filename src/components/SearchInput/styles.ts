import styled from "styled-components";

export const InputContainer = styled.div`
  border: 1px solid #5f6368;
  border-radius: 24px;
  display: flex;
  justify-content: center;
  place-items: center;
  width: 60%;
  height: 40px;
  margin: 0 20%;
`;

export const Input = styled.input`
  margin: 0 auto;
  border: none;
  border-radius: 24px;
  height: 38px;
  width: 95%;
  font-size: 18px;
  :focus {
    outline-width: 0;
  }
`;
