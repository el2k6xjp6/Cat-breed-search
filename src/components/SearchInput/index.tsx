import React from "react";
import { InputContainer, Input } from "./styles";
import SearchIcon from "../SearchIcon";

interface Props {
  queryString: string;
  handleQueryInput: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

function SearchInput(props: Props) {
  return (
    <InputContainer>
      <SearchIcon />
      <Input
        id="input"
        onChange={props.handleQueryInput}
        value={props.queryString}
        type="text"
        placeholder="The name, or part of the name."
      />
    </InputContainer>
  );
}

export default React.memo(SearchInput);
