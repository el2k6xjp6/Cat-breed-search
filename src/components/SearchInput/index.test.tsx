import { shallow } from "enzyme";
import SearchInput from ".";

import "jest-styled-components";
import renderer from "react-test-renderer";

const mockProps = {
  queryString: "",
  handleQueryInput: jest.fn(),
};

describe("test SearchInput", () => {
  test("Snapshot for SearchInput", () => {
    const tree = renderer.create(<SearchInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("input", () => {
    const wrapper = shallow(<SearchInput {...mockProps} />);
    wrapper.find("[id='input']").simulate("change", "a");
    expect(mockProps.handleQueryInput).toHaveBeenCalledTimes(1);
    expect(mockProps.handleQueryInput).toBeCalledWith("a");
  });
});
