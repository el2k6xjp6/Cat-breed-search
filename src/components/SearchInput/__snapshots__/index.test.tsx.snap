// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchInput Snapshot for SearchInput 1`] = `
.c0 {
  border: 1px solid #5f6368;
  border-radius: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  place-items: center;
  width: 60%;
  height: 40px;
  margin: 0 20%;
}

.c2 {
  margin: 0 auto;
  border: none;
  border-radius: 24px;
  height: 38px;
  width: 95%;
  font-size: 18px;
}

.c2:focus {
  outline-width: 0;
}

.c1 {
  height: 30px;
  width: 30px;
  margin-left: 10px;
}

<div
  className="c0"
>
  <svg
    className="c1"
  >
    <circle
      cx="12"
      cy="12"
      fill="white"
      r="8"
      stroke="black"
      strokeWidth="3"
    />
    <line
      stroke="black"
      strokeWidth="3px"
      x1="18"
      x2="26"
      y1="18"
      y2="26"
    />
  </svg>
  <input
    className="c2"
    id="input"
    onChange={[MockFunction]}
    placeholder="The name, or part of the name."
    type="text"
    value=""
  />
</div>
`;
